knotx-core/src/main/java/io/knotx/knot/AbstractKnotProxy.java

```diff
@@ -72,6 +73,16 @@ public void process(KnotContext knotContext, Handler<AsyncResult<KnotContext>> r
   /**
    * Method lets you decide whether the Fragment should be processed by your Knot or not.
    *
+   * @param fragment Fragment to process - or skip  {@link Fragment}.
+   * @return <tt>true</tt> if this Knot should process current {@link Fragment}.
+   */
+  protected boolean shouldProcess(Fragment fragment) {
```

Comment:
Let's mark `shouldProcess(Set<String> knots)` as `@Deprecated`. Method you suggested should replace it.

Reply:
moved here: https://github.com/Cognifide/knotx/issues/474